modules/weather/apixu/apixu.go

```diff
@@ -0,0 +1,179 @@
+package apixu // import "barista.run/modules/weather/apixu"
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"barista.run/modules/weather"
+
+	"github.com/martinlindhe/unit"
+)
+
+// Config represents apixu API configuration (just the API key)
+// from which a weather.Provider can be built.
+type Config string
+
+// New creates a new Apixu API configuration.
+func New(apiKey string) Config {
+	return Config(apiKey)
+}
+
+// Query queries Apixu using zip code, lat/lon, city name, etc. (see https://www.apixu.com/doc/request.aspx)
+func (c Config) Query(q string) weather.Provider {
+	qp := url.Values{}
+	qp.Add("key", string(c))
+	qp.Add("q", q)
+	apixuURL := url.URL{
+		Scheme:   "http",
+		Host:     "api.apixu.com",
+		Path:     "/v1/current.json",
+		RawQuery: qp.Encode(),
+	}
+	return Provider(apixuURL.String())
+}
+
+// Provider wraps an Apixu API url so that
+// it can be used as a weather.Provider.
+type Provider string
+
+// apixuWeather represents an Apixu json response.
+type apixuWeather struct {
+	Location struct {
+		Name           string  `json:"name"`
+		Region         string  `json:"region"`
+		Country        string  `json:"country"`
+		Lat            float64 `json:"lat"`
+		Lon            float64 `json:"lon"`
+		TzID           string  `json:"tz_id"`
+		LocaltimeEpoch int64   `json:"localtime_epoch"`
+		Localtime      string  `json:"localtime"`
+	} `json:"location"`
+
+	Current struct {
+		Condition struct {
+			Text string `json:"text"`
+			Icon string `json:"icon"`
+			// Code is a code for describing weather (see https://www.apixu.com/doc/weather-conditions.aspx)
+			Code int `json:"code"`
+		} `json:"condition"`
+
+		LastUpdated      string `json:"last_updated"`
+		LastUpdatedEpoch int64  `json:"last_updated_epoch"`
+
+		TempC float64 `json:"temp_c"`
```

Comment:
Here and elsewhere, don't keep properties in this struct that aren't used to create the weather.
e.g. only TempF is required, so you can remove TempC. (Similarly WindKPH, PressureIN, etc.)